pkg/operator/defaultscccontroller/defaultscc.go

```diff
@@ -0,0 +1,98 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/openshift/api"
+	securityv1 "github.com/openshift/api/security/v1"
+
+	assets "github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/defaultscc_assets"
+)
+
+// DefaultSCCCache maintains the list of default SecurityContextConstraints objects
+// shipped wth an OpenShift cluster.
+type DefaultSCCCache struct {
+	orderedNames []string
+	set          map[string]*securityv1.SecurityContextConstraints
+}
+
+// Get returns the SecurityContextConstraints object associated with the
+// specified name. The returned SecurityContextConstraints is a deep copy and so
+// it can be modified at will without impacting the integrity of the underlying cache.
+func (d *DefaultSCCCache) Get(name string) (scc *securityv1.SecurityContextConstraints, exists bool) {
+	obj, exists := d.set[name]
+	if !exists {
+		return
+	}
+
+	scc = obj.DeepCopy()
+	return
+}
+
+func (d *DefaultSCCCache) DefaultSCCNames() []string {
+	return d.orderedNames
+}
+
+// NewDefaultSCCCache renders the assets associated with the default set of SCC and
+// loads it into a cache (DefaultSCCCache)
+func NewDefaultSCCCache() (cache *DefaultSCCCache, err error) {
+	decoder, decoderErr := decoder()
+	if decoderErr != nil {
+		err = fmt.Errorf("failed to create decoder - %s", decoderErr.Error())
+		return
+	}
+
+	names := make([]string, 0)
+	set := make(map[string]*securityv1.SecurityContextConstraints)
+	for _, name := range assets.AssetNames() {
+		bytes, assetErr := assets.Asset(name)
+		if assetErr != nil {
+			return
+		}
+
+		object, _, decodeErr := decoder.Decode(bytes, nil, nil)
+		if decodeErr != nil {
+			err = fmt.Errorf("failed to decode SecurityContextConstraints from asset name=%s - %s", name, decodeErr.Error())
+			return
+		}
+
+		scc, ok := object.(*securityv1.SecurityContextConstraints)
+		if !ok {
+			err = fmt.Errorf("obj is not SecurityContextConstraint type name=%s", name)
+			return
+		}
+
+		_, exists := set[scc.GetName()]
+		if exists {
+			err = fmt.Errorf("SecurityContextConstraint already exists in set name=%s", scc.GetName())
+			return
+		}
+
+		set[scc.GetName()] = scc
+		names = append(names, scc.GetName())
+	}
+
+	cache = &DefaultSCCCache{
+		orderedNames: names,
+		set:          set,
+	}
+	return
+}
+
+func decoder() (decoder runtime.Decoder, err error) {
+	scheme := runtime.NewScheme()
+	if err = api.InstallKube(scheme); err != nil {
+		return
+	}
+	if err = api.Install(scheme); err != nil {
```

Comment:
installing all into the scheme is not necessary. Install the security group only.